src/vs/platform/log/common/log.ts

```diff
@@ -147,7 +147,7 @@ export interface ILoggerService {
 	 *
 	 * This will also register the logger with the logger service.
 	 */
-	createLogger(resource: URI, options?: ILoggerOptions): ILogger;
+	createLogger(resource: URI, options?: ILoggerOptions, donotRegister?: boolean): ILogger;
```

Comment:
Maybe `doNotRegister` is better?

Though I would prefer a method `createDetachedLogger`.

Reply:
This argument is temporary until this is adopted https://github.com/microsoft/vscode/issues/172031 - I would remove donotRegister argument after.

Reply:
I think it might be helpful to indicate "temporary" with a comment :)

Reply:
I agree. 

But, I am resisting this change and adopt the above item now itself so that we do not need this workaround.